### ISSUE TITLE
Key Value fields can cause a ES Mapping Explosion

### ISSUE THREAD

**Issue by wezell** [dotCMS/core#18446]
When a Key/Value field is added to dotCMS, it is automatically set to "indexed" and there is no way to disable this.  The issue is that the way we index key value fields can/will lead to a mapping explosions as we are indexing each key as a field in elasticsearch

https://www.elastic.co/blog/found-crash-elasticsearch#mapping-explosion

It seems that even if we indexed these as nested fields, it would still cause the issue:

https://discuss.elastic.co/t/can-nested-fields-prevent-mapping-explosion/95464

1. We need to allow a customer to enable/disable "Is indexed" on key/value fields
2. We might need to index key/values as a text or as a keyword array of Strings, like `key=property`, e.g. `["Key 1=My first value","Key two=another key","final key=here is a big fat value"]`
3. Binary Metadata still needs to make it into the index and be searchable as it has been in the past (meaning, a field for each key, e.g. `fileAsset.metaData.fileName`, `fileAsset.metaData.author`

**Comment by wezell** [comment]
Workaround for now
```
update field set searchable=false, unique_=false, listed=false, indexed=false  where field_type='com.dotcms.contenttype.model.field.KeyValueField' and velocity_var_name!='metaData'
```

**Comment by fabrizzio-dotCMS** [comment]
PR https://github.com/dotCMS/core/pull/18503

**Comment by alfredo-dotcms** [comment]
Tested on master with master generated data, when looking in kibana for `working` index, the result shows the content with the new structure
![image](https://user-images.githubusercontent.com/37185433/82258140-881aa800-9916-11ea-8713-1ab5d72390e3.png)


also when using the `Query Tool` you can see the content generated
![image](https://user-images.githubusercontent.com/37185433/82258234-b304fc00-9916-11ea-81f5-ea6e6ee85862.png)

**Comment by fabrizzio-dotCMS** [comment]
I was finally able to test this fix against the customer's database where the issue originated.
Aske me about it when Testing.
 
I had to increase my docker resources like this:
Memory: 11.00 GB
Swap: 1.5 GB
Disk Image: 112 GB
Otherwise, my Elasticsearch was getting me: java.io.IOException: No space left on device
Once I bumped up my resources in docker I fired a reindex.
Then I used the following query to determine what content-types had a key-value field

```
select s.name, f.field_name, f.field_type from field f join structure s on f.structure_inode = s.inode
where f.field_type='com.dotcms.contenttype.model.field.KeyValueField' 
and f.velocity_var_name<>'metaData'
```

The reindex ran for a couple of hours and would complete successfully,
Except for 1 record that had a Malformed JSON stored in the DB. It was missing a closing double quote `"`. Once I got that fixed in the database. The reindex completed smoothly. 

Indices On Kibana showing the key-value.
 
![Screen Shot 2020-05-20 at 08 42 18](https://user-images.githubusercontent.com/37148755/82460233-28d3a980-9a76-11ea-9b21-50eed0853237.png)

Actual Contentlet indexed loaded in dotCMS.

![Screen Shot 2020-05-20 at 08 41 29](https://user-images.githubusercontent.com/37148755/82460264-312be480-9a76-11ea-8b29-1e93b4efc0a5.png)

**Comment by fabrizzio-dotCMS** [comment]
**Note to QA** Please validate the Lucene queries against the customer data:
```
 +taginformation.bannerbutton.key:Ask A Student* 
 +taginformation.bannerbutton.value:http*
```

**Comment by wezell** [comment]
This should work, please test
`+taginformation.bannerbutton.key_dotraw:Ask A Student`

**Comment by nollymar** [comment]
We should re-test this issue using flattened fields instead and customer data. Further info:
https://www.elastic.co/guide/en/elasticsearch/reference/master/flattened.html
https://www.elastic.co/guide/en/elasticsearch/reference/master/mapping.html#mapping-limit-settings

**Comment by nollymar** [comment]
The original PR was restored here: https://github.com/dotCMS/core/pull/18687

**Comment by nollymar** [comment]
**Passed internal qa:** This change was tested using custom data that previously failed for the same reason

**Comment by bryanboza** [comment]
Fixed, tested on release-5.3.3 // Postgres // FF
